internal/users/controller/auth.go

```diff
@@ -0,0 +1,41 @@
+package controller
+
+import (
+	"net/http"
+
+	"github.com/KnoblauchPilze/user-service/internal/users/service"
+	"github.com/KnoblauchPilze/user-service/pkg/errors"
+	"github.com/KnoblauchPilze/user-service/pkg/middleware"
+	"github.com/KnoblauchPilze/user-service/pkg/rest"
+	"github.com/labstack/echo/v4"
+)
+
+func AuthEndpoints(service service.AuthService) rest.Routes {
+	var out rest.Routes
+
+	authHandler := fromAuthServiceAwareHttpHandler(authUser, service)
+	auth := rest.NewRoute(http.MethodGet, false, "/users/auth", authHandler)
+	out = append(out, auth)
+
+	return out
+}
+
+func authUser(c echo.Context, as service.AuthService) error {
```

Comment:
The new endpoint: this takes an API key from the request passed as a header and tries to match it with a valid API key registered in the database. If it works, it returns the ACLs and the permissions as the body of the response. Otherwise it returns an appropriate error code (authentication expired, no such user, etc.).